### PR TITLE
feat(analytics): add analytics on noResult and documentOpen

### PR DESCRIPTION
**Pull request by aiheon**
### Proposed Changes

https://coveord.atlassian.net/browse/UITOOL-402

Adding analytics on No result after using the search bar and on document open in the search results.

More analytics will come in another pr when I change the search bar for the Atomic one :eyes: 

### Potential Breaking Changes

None

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/website/src/pages/Search.tsx [hunk 1]
```diff
@@ -44,13 +46,14 @@ const ResultListRenderer: FunctionComponent<ResultListProps> = (props) => {
                             {state.results.length === 1 ? ' result' : ' results'}
                         </p>
                         <div className="tile-grid">
-                            {state.results.map(({title, raw, uniqueId, clickUri}: Result) => (
+                            {state.results.map((result) => (
```
Comment (toofff) [inline packages/website/src/pages/Search.tsx:49]: Why did you remove the typing?

**Review by gdostie — APPROVED** [review]

File: packages/website/src/search/NoSearchResultTemplate.tsx [hunk 2]
```diff
@@ -14,6 +14,11 @@ interface NoResultTemplateProps {
 export const NoSearchResultTemplate: FunctionComponent<NoResultTemplateProps> = (props) => {
     const {engine, query} = props;
 
+    const historyManager: HistoryManager = buildHistoryManager(engine);
+    const [state, setState] = useState(historyManager.state);
```
Comment (toofff) [inline packages/website/src/search/NoSearchResultTemplate.tsx:18]:
```suggestion
    const [state, setState] = useState<HistoryManagerState>(historyManager.state);
```

**Review by toofff — APPROVED** [review]

File: packages/website/src/pages/Search.tsx (hunk 1 above)
Comment (aiheon) [reply packages/website/src/pages/Search.tsx:49]: Because I need the result object for the log :smile: 
https://docs.coveo.com/en/headless/0.28.1/reference/actions/click-analytics/#logdocumentopen

**Comment by aiheon** [comment]
oops :see_no_evil:
